ibis/backends/flink/registry.py

```diff
@@ -219,6 +268,52 @@ def _floor_divide(translator: ExprTranslator, op: ops.Node) -> str:
     return f"FLOOR(({left}) / ({right}))"
 
 
+def _array_index(translator: ExprTranslator, op: ops.arrays.ArrayIndex):
+    table_column = op.arg
+    index = op.index
+
+    table_column_translated = translator.translate(table_column)
+    index_translated = translator.translate(index)
+
+    return f"{table_column_translated} '[' {index_translated} ']'"
```

Comment:
Can you point to the place where this syntax is defined? It looks a little strange to my eye, like an array column would render as `a['0']` or `a['expr']` in the case of a more complex `expr`. Is that correct?

Reply:
> Can you point to the place where this syntax is defined?

It is defined [here](https://nightlies.apache.org/flink/flink-docs-release-1.13/docs/dev/table/functions/systemfunctions/#collection-functions) as `array '[' INT ']'`.

> It looks a little strange to my eye

I agree :)

> like an array column would render as a['0'] or a['expr'] in the case of a more complex expr.

It would look like `array'['0']'`. It is only defined for an integer index.

Reply:
Ah, got it.

I read the docs for the operation and I think you need to add ` + 1` to the index, because arrays are zero-indexed in ibis.

Reply:
I'm pretty sure these embedded quotes are incorrect. I understand that the Flink documentation uses this notation.

However, this notation is often how parser-generator grammars for this operation are written, and I suspect that `'['` was copy-pasted over directly from the parser and that

```
x'['0']'
```

is definitely not a valid Flink SQL expression.

@mfatihaktas @deepyaman @chloeh13q Can someone validate this?

Reply:
https://docs.confluent.io/cloud/current/flink/reference/functions/collection-functions.html#flink-sql-array-collection is an example of a place where an example uses `array_name[index]`.

Reply:
Thanks, you are right, this caused an error and I fixed it now. The problem earlier was that the test function(s) that I thought were validating this operator were actually not covering the operator.